### ISSUE TITLE
Code Path Analysis cannot handle LabeledStatement with other than switches or loops.

### ISSUE THREAD

**Issue by mysticatea** [eslint/eslint#5171]
I had not known it at that time.
This causes some false positive of rules which are using code path (e.g. `no-unreachable`).

```js
A: {
    break A;
}

foo(); // Unreachable code. (no-unreachable) <<< false positive
```

I will fix this.

**Comment by mysticatea** [comment]
[This foolproof code](https://github.com/eslint/eslint/blob/master/lib/code-path-analysis/code-path-state.js#L1202-L1206) has prevented crash :D
